lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -363,6 +363,12 @@ def _prepare_environment(  # noqa: C901
                     else:
                         existing_location = node_runbook.location
 
+                    if node_runbook.marketplace:
```

Comment:
I think it again, it should be put before the last return, and in the `if environment.runbook.nodes_requirement:`. There are some different scenarios. That's the right place to resolve the image name. You can add a new for loop there.

Reply:
Do you mean just before `return is_success` on line 482?

Reply:
yes